include/permanent.hpp

```diff
@@ -136,21 +134,22 @@ inline T permanent(std::vector<T> &mat) {
         for (llint k = threadbound_low[ii]; k < threadbound_hi[ii]; k++) {
             T rowsumprod = static_cast<T>(1);
             llint kg2 = igray(k+1);
-            llint  sgntmp = kg2 - igray(k);
-            llint  sig = sgntmp/std::abs(sgntmp);
-            int  pos = 0;
-
-            pos = n - left_most_set_bit(sgntmp);
+            llint sgntmp = kg2 - igray(k);
+            llint sig = sgntmp/std::abs(sgntmp);
```

Comment:
Just a query: for logic using Gray codes, does the use of a signed integer make sense? Would an unsigned value be a better choice here?

Reply:
I also think that correct way to do this is to use `unsigned`, and I was going to implement it that way but before doing it I reviewed the current logic within the code base, which uses `signed`, and for the sake of consistency I used the same logic. But now that you mentioned it I think we should fix all.

Reply:
For consistency, if these changes have not yet been made, I think it is fine to keep the existing implementation. We can always tackle these in a later PR across the package. 

If they have already been moved to unsigned then all good too.